### ISSUE TITLE
How to disable vcs.url ?

### ISSUE THREAD

**Issue by gsemet** [jfrog/jfrog-cli#643]
Hello

On my `rt upload` I see some properties automatically set, especially `vcs.url`. This URL may contains sensitive information (username/password). Can you hide the password part in the uploaded url (there is no reason to EVER upload this credentials information) or disable setting these properties by default?

Thanks

**Comment by eyalbe4** [comment]
@gsemet,
Using the latest release of JFrog CLI should properly mask the credentials part in the *vcs.url*. Which version are you using? If it is not the latest release, can you please upgrade and let us know if the issue is resolved?
Thanks

**Comment by eyalbe4** [comment]
@gsemet,
Were you able to resolve the issue by upgrading JFrog CLI?

**Comment by gsemet** [comment]
indeed, but I have to remove all previously uploaded properties with leaked credentials manually :(

**Comment by eyalbe4** [comment]
@gsemet,
We truely apologise for this incident.
I believe that you can clean these up easily using the *jfrog rt set-props* command.
Please let me know if you need any assistance with this.

**Comment by gsemet** [comment]
oh, I was going to write a script to do it... does the "jfrog rt set-props" can delete properties or only overwrite it?

**Comment by eyalbe4** [comment]
@gsemet,
I actually meant to suggest the *jfrog rt delete-props* command. It will complete delete the properties sent to it.

**Comment by serGrey** [comment]
I updated to version **1.35.3**
**vcs.url** & **vcs.revision** automatically set and I received the error:
 **404 - File or directory not found**
After investigation, I found the reason: my git repo url (in the git config) contains **space**
Could you please help me, how I can exclude vcs.url properties or replace \ escape the **space**?

**Comment by eyalbe4** [comment]
@serGrey,
Thank you for reporting this!
We're looking into this issue now.

**Comment by serGrey** [comment]
Thank you
Is it possible to exclude or escape **vcs.url**?
Or I should try to revert extension JFrog Artifactory with Visual Studio Team Services to the previous version, 1.8.9?
Currently, all builds with **space** in the git repo url - failed

**Comment by eyalbe4** [comment]
@serGrey,
Can you temporarily disable the *Collect build-info* option in the tasks which fail? This will disable the vcs properties from being added. We'll let you know as soon as the patch with the fix is released, so that you can enable it again. 
Please let us know this works for now.

**Comment by serGrey** [comment]
I disabled option _Collect build info_
It's working.
Thank you

**Comment by serGrey** [comment]
But now I have another issue when I disable option _Collect build info_ : artifact not visible in the Builds after upload, just in the local repository

**Comment by eyalbe4** [comment]
@serGrey - 

The fact that you don't see the artifacts as part of the build is expected, because we disabled the build-info collection (by disabling the *Collect build info* in the task).

I'm actually not able to reproduce the issue you described. Everything seems to be working for me, even with a white-space in the git URL.
Here's the content of my **config** file under the local **.git** directory:
```
[core]
	repositoryformatversion = 0
	filemode = true
	bare = false
	logallrefupdates = true
	ignorecase = true
	precomposeunicode = true
[remote "origin"]
	url = https://github.com/jfrog/jfrog cli.git
	fetch = +refs/heads/*:refs/remotes/origin/*
[branch "master"]
	remote = origin
	merge = refs/heads/master
[branch "dev"]
	remote = origin
	merge = refs/heads/dev
```
I used the local jfrog-cli project, which I cloned from ```https://github.com/jfrog/jfrog-cli.git``` After cloning it, I manually edited the above config file, and added a white-space here --> ```...jfrog cli.git```
While inside the *jfrogt-cli* directory, I ran:
```
jfrog rt u "m*.go" generic-local --build-name test --build-number 1
```
And 12 files where successfully uploaded:
```
 Log path: /Users/eyalb/.jfrog/logs/jfrog-cli.2020-04-02.18-47-23.21809.log
{
  "status": "success",
  "totals": {
    "success": 12,
    "failure": 0
  }
}
```
If I understand correctly, this is where you get the error.
Searching one of the files after the upload, also returns the **vcs.url** as expected (with the white-space):
```
jfrog rt s generic-local/main.go

[Info] Searching artifacts...
[Info] Found 1 artifact.
[
  {
    "path": "generic-local/main.go",
    "type": "file",
    "size": 3015,
    "created": "2020-04-02T18:13:16.656+03:00",
    "modified": "2020-04-02T18:47:24.152+03:00",
    "props": {
      "build.name": [
        "test"
      ],
      "build.number": [
        "1"
      ],
      "build.timestamp": [
        "1585842444043"
      ],
      "vcs.revision": [
        "54a1635561b558da866e002792a86f9d148782e6"
      ],
      "vcs.url": [
        "https://github.com/jfrog/jfrog cli.git"
      ]
    }
  }
]
```
Will you able to try what I did (with the same jfrog-cli project) and see if this works for you or not? If the above works, can you try to compare the **.git** that is failing with this one?
Please let me know what you see.

**Comment by serGrey** [comment]
You right, when I run locally in case specify files to upload - it's worked fine
But when I try to use spec file - got the error
Example:

**worked without error**:
jfrog.exe rt u pkg/artf.zip artf_local_repo --build-name=TestBuild --build-number=2

**with error**:
jfrog.exe rt u --spec=spec.json artf_local_repo --build-name=TestBuild --build-number=2

**with error**
`[core]
	repositoryformatversion = 0
	filemode = false
	bare = false
	logallrefupdates = true
	symlinks = false
	ignorecase = true
[remote "origin"]
	url = https://github.test/repo/repowithspace/repo.git
	fetch = +refs/heads/*:refs/remotes/origin/*
[branch "master"]
	remote = origin
	merge = refs/heads/master
`

**without error**
`[core]
	repositoryformatversion = 0
	filemode = false
	bare = false
	logallrefupdates = true
	symlinks = false
	ignorecase = true
[remote "origin"]
	url = https://github.test/repo/repowithspace/repo.git
	fetch = +refs/heads/*:refs/remotes/origin/*
[branch "master"]
	remote = origin
	merge = refs/heads/master
`

**Comment by eyalbe4** [comment]
@serGrey,

Can you please try running the upload command without a file spec again, but this time, upload multiple files? In your previous attempt, you uploaded only one file - artf.zip.
There's currently a bug in JFrog CLI, which causes it not to add the **vcs.url** property, when there's only one file uploaded. Try using a wildcard in the upload command, so that it uploads more than one file. Let me know if this error reproduces even without a file spec in that case.

I'm unable to reproduce the error when multiple files are uploaded, with or without file spec. Does the error reproduce for you with the jfrog-cli repository I used above (after adding the white-space manually in the config)? If it does not, then we'll need to compare the working and not working .git directories, to find out what causes this issue.

**Comment by serGrey** [comment]
@eyalbe4,
I confirm, repo url with **space** - is not the main reason.
It's reproducible:
**with error:**
**git repo url** = https://github.test/repo/repo with space/repo.git
jfrog.exe rt u --spec=spec.json artf_local_repo --build-name=TestBuild --build-number=2
--spec=spec.json
```
{
  "files": [
    {
      "pattern": "c:/test/dist/(.*).zip",
      "target": "artf_builds/TestBuild/{1}",
      "flat": "true",
      "recursive": "true",
      "regexp": "true"
    }
  ]
}
```

**with error:**
**git repo url** = https://github.test/repo/repo with space/repo.git
jfrog.exe rt u c:/test/dist/*.zip artf_local_repo --build-name=TestBuild --build-number=2
--spec=spec.json



**without error:**
**git repo url** = https://github.test/repo/repowithspace/repo.git
jfrog.exe rt u --spec=spec.json artf_local_repo --build-name=TestBuild --build-number=2
--spec=spec.json
```
{
  "files": [
    {
      "pattern": "c:/test/dist/(.*).zip",
      "target": "artf_builds/TestBuild/{1}",
      "flat": "true",
      "recursive": "true",
      "regexp": "true"
    }
  ]
}
```


**without error:**
**git repo url** = https://github.test/repo/repo with space/repo.git
jfrog.exe rt u --spec=spec.json artf_local_repo --build-name=TestBuild --build-number=2
--spec=spec.json
```
{
  "files": [
    {
      "pattern": "c:/test/dist/artf.zip",
      "target": "artf_builds/TestBuild/",
      "flat": "true",
      "recursive": "true",
      "regexp": "true"
    }
  ]
}
```

**without error:**
**git repo url** = https://github.test/repo/repo with space/repo.git
jfrog.exe rt u c:/test/dist/artf.zip artf_local_repo --build-name=TestBuild --build-number=2
--spec=spec.json


**without error:**
**git repo url** = https://github.test/repo/repowithspace/repo.git
jfrog.exe rt u c:/test/dist/artf.zip artf_local_repo --build-name=TestBuild --build-number=2
--spec=spec.json




**without error:**
**git repo url** = https://github.test/repo/repowithspace/repo.git
jfrog.exe rt u c:/test/dist/*f.zip artf_local_repo --build-name=TestBuild --build-number=2
--spec=spec.json

**Comment by eyalbe4** [comment]
@serGrey,
Were you able to determine what causes the issue?

**Comment by serGrey** [comment]
@eyalbe4,
Combination: git repo with **space** and specify multiple files (with mask), with or without spec file

**Comment by danno131313** [comment]
I am seeing this issue as well in 1.38, but my git repo URL doesn't include any spaces and I am only uploading a single file. Running the upload commands manually works, but through Gitlab CI on a runner it does not. We downgraded to 1.31.2 and no longer see the 404 - not found.

**Comment by eyalbe4** [comment]
@danno131313,
Can you please share the exact JFrog CLI command which fails, including all the arguments and options?
Please also share all the log prints.
Also, can you please share your .git directory?
This should help us understand the root cause of the issue.

**Comment by serGrey** [comment]
the same result in case buildName or buildNumber contains at least one space
jfrog cli : 1.41

**Comment by gbaychev** [comment]
Hello, I know it is almost a year later, but I ran in the same problems and I want to share some findings.
First of all, I'm using
- jfrog-cli, 1.47.1, but the same happens with 2.0
- The bitnami prepackaged instance on AWS with Artifactory OSS 7.7

As long as there is an urlencoded character in vcs.url, I get 404. If the characters are not encoded, everything is fine.
Without error (unauthorized is fine in this situation)

![image](https://user-images.githubusercontent.com/3358056/133636283-ed4353da-0932-49bf-97dc-8b630caa7e85.png)

With error:

![image](https://user-images.githubusercontent.com/3358056/133636978-6c8029ed-4279-4df8-bff0-8d398e1b3485.png)

What is also interesting, is that I see the HTTP 401 request in artifactory-request.log, but do not see the 404 one. The 404 one can be found only in the apache log.

Out of curiosity: why do you try to stuff all this information in the uri? Wouldn't be much better to send it in the response body?

The obvious workaround (at least for me) is to 'collect' the build information in a folder, which does not contain a .git folder. But I do not need the revision or the git remote in the build info, which is not everybody's case, I guess.

**Comment by gbaychev** [comment]
Yeah, I got it. It was apache config issue. I needed to add `AllowEncodedSlashes On` at the end of httpd.conf and to each VirtualHost.
In case of bitnami, the VirtualHosts can be found at: /opt/bitnami/apache2/conf/bitnami/httpd.conf

**Comment by shuvadipc** [comment]
Not relevant. Closing this.
